report-jtreg-comparator/src/main/java/io/jenkins/plugins/report/jtreg/main/comparator/StackTraceCompare.java

```diff
@@ -0,0 +1,167 @@
+
+            // create a list of values where to put the percentages
+            List<String> putList = new ArrayList<>();
+            for (Map.Entry<String, ArrayList<String>> entry : failedMap.entrySet()) {
+                if (entry.getValue().contains(test)) {
+                    putList.add(entry.getKey());
+                }
+            }
+
+            Collections.sort(putList); // just to be sure
+
+            // get the referential stack trace
+            String reference = null;
+            if (options.getReferentialJobName() == null || options.getReferentialBuildNumber() == -1) {
+                // if not set, get the first one in table as referential
+                reference = getTestTrace(new File(putList.get(0)), test);
+            } else {
+                String value = putList.stream()
+                        .filter(build -> build.matches(".*jobs/" + options.getReferentialJobName() + "/builds/" + options.getReferentialBuildNumber()))
+                        .findFirst()
+                        .orElse(null);
+
+                if (value != null) {
+                    reference = getTestTrace(new File(value), test);
+                }
+
+            }
+
+            // calculate and put the percentages (or - if no referential was found)
+            for (String value : putList) {
+                String stringToPut = "-";
+
+                if (reference != null) {
+                    String second = getTestTrace(new File(value), test);
+                    stringToPut = String.valueOf(getTraceSimilarity(reference, second));
+                }
+
+                table[i][jobBuilds.indexOf(value) + 1] = stringToPut;
+            }
+
+            // TODO delete, just for debug logging
+            System.err.println("Test " + (i - 1) + "/" + failedTests.size() + " - " + (int)((i - 1)/(double)failedTests.size() * 100) + "%");
+
+            i++;
+        }
+
+        // print the table into stdout
+        options.getFormatter().printTable(table, failedTests.size() + 1, jobBuilds.size() + 1);
+    }
+
+    private static String getTestTrace(File build, String testName) {
+        // TODO:
+        // - for now, it only takes system-out trace, it should also work with system-err (maybe combine them?)
+        // - cut the trace shorter, it its too big (first or last 100kb maximum - probably a new switch)
+
+        try {
+            BuildReportExtended bex = bs.parseBuildReportExtended(new RunWrapperFromDir(build), null);
+
+            // go through the suites
+            for (Suite s : bex.getSuites()) {
+                // filter for the test with the name we are looking for
+                Test t = s.getReport().getTestProblems().stream().filter(test -> test.getName().equals(testName)).findFirst().orElse(null);
+                if (t != null) {
+                    // return the output (stack trace)
+                    return t.getOutputs().get(0).getValue();
```

Comment:
As we spoke on meeting. Thisis more complicated.
We ahve three options: 
1 - join statusLine and all t.getOutputs() to one huge string,. print one number
2 - comapre adequate get.value() (where adequate is equality  of get.getName - not the order; if no adequate, then 0%  match). Status l;ine can be coonsidered as special key.
  2.1 - make avarage od vlaues, and print one number
  2.2 - print one N% numbers, each per key comparison.

Note that in case of future diff trace service, this may be tricky.

Reply:
in html, would be nice to have tooltip, which number is what.